RevenueCatUI/Views/RevenueCatDeeplinkHandler.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  RevenueCatDeeplinkHandler.swift
+//
+//  Created by Andrés Boedo on 9/17/24.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable, message: "RevenueCatUI does not support macOS yet")
+@available(tvOS, unavailable, message: "RevenueCatUI does not support tvOS yet")
+struct RevenueCatDeeplinkHandlerView<Content: View>: View {
+    @State private var isShowingPaywall: Bool = false
+    @State private var offeringID: String?
+
+    let content: Content
+
+    init(@ViewBuilder content: () -> Content) {
+        self.content = content()
+    }
+
+    var body: some View {
+        content
+            .onOpenURL { url in
+                if let extractedOfferingID = extractOfferingID(from: url) {
```

Comment:
Hmm correct me if I'm wrong but this would actually open in any deep link that has the `offeringID` query parameter right? I wonder if we should verify the deep links path as well? As in, having `anything://paywall?offeringID=test-offering-id` and we verify the path is `/paywall` and the `offeringID` exists.

Reply:
Oh I see you mentioned this in another comment. Ignore me :P

Reply:
haha yeah I'm not validating, this is as quick and dirty as it comes